addons/website_sale/controllers/main.py

```diff
@@ -185,6 +182,10 @@ def _get_search_domain(self, search, category, attrib_values, search_in_descript
             if attrib:
                 domains.append([('attribute_line_ids.value_ids', 'in', ids)])
 
+
+        if not request.env.user._is_admin():
```

Comment:
why ? probably not

Reply:
Else unpublished products are taken into account for the total per categories on a regular user

Reply:
so it should be done into the count, not the select...